graphql/src/main/java/com/scalar/db/graphql/server/GraphQlHandler.java

```diff
@@ -0,0 +1,258 @@
+package com.scalar.db.graphql.server;
+
+import com.google.common.io.ByteStreams;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+import com.scalar.db.config.DatabaseConfig;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.graphql.GraphQlFactory;
+import com.scalar.db.service.StorageFactory;
+import com.scalar.db.service.TransactionFactory;
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.nio.charset.Charset;
+import java.nio.charset.IllegalCharsetNameException;
+import java.nio.charset.StandardCharsets;
+import java.nio.charset.UnsupportedCharsetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+
+class GraphQlHandler extends AbstractHandler {
+  private static final Pattern CONTENT_TYPE_JSON =
+      Pattern.compile(
+          "^application/(?:graphql\\+)?json\\s*(?:;\\s*charset=(.+))?", Pattern.CASE_INSENSITIVE);
+  private static final Pattern CONTENT_TYPE_GRAPHQL =
+      Pattern.compile("^application/graphql\\s*(?:;\\s*charset=(.+))?", Pattern.CASE_INSENSITIVE);
+  private static final Gson GSON =
```

Comment:
One (currently minor) concern to be discussed maybe; Gson might be slow.
https://github.com/fabienrenaud/java-json-benchmark
Thus, maybe, better to have an abstraction to be able to change the implementation later on.

I'm working on defining a new interface for Contract in Scalar DL,
and one of the major reasons is `javax.json` is super slow and it starts to become too much overhead.
Each Scalar DB API request is more lightweight than Scalar DL, so Gson overhead might stand out more.

Reply:
I didn't know that there were that many JSON implementations in Java! I chose GSON to reuse the dependency that the schema-loader already used.
Serializing and deserializing with GSON is very simple (`GSON.toJson` and `GSON.fromJson`), so it would be easy to switch the implementation later. But I added a small refactoring to make a GSON-related variable encapsulated in a method, i.e., `JsonObject` no longer appears as an argument of the method: 6ccc1c4